src/snowflake/connector/cursor.py

```diff
@@ -911,6 +926,36 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """Context manager with commit or rollback."""
         self.close()
 
+    def get_results_from_sfqid(self, sfqid: str):
+        """Gets the results from previously ran query."""
+
+        def wait_until_ready():
+            """Makes sure query has finished executing and once it has retrieves results."""
+            status = self.connection.get_query_status(sfqid)
+            no_data_counter = 0
+            no_data_max_retry = 24
+            while self.connection.is_still_running(status):
+                status = self.connection.get_query_status(sfqid)
+                if not self.connection.is_still_running(status) and status != QueryStatus.SUCCESS:
+                    raise DatabaseError(
```

Comment:
If this is an acceptable scenario do we really want to raise an error?

Reply:
I have copied this error from the JDBC PR.
But this would be triggered in the following case, you started a query and tried to get results while it was still running. Then you waited and all of the sudden it was done, but failed with an error.
I think raising an error in this case would be acceptable.